src/univaf_data.py

```diff
@@ -0,0 +1,79 @@
+from contextlib import contextmanager
+import gzip
+import json
+import os
+from pathlib import Path
+import subprocess
+import sys
+import urllib.request
+
+UNIVAF_AWS_BUCKET = 'univaf-data-snapshots'
+DATA_PATH = Path(__file__).parent.parent.absolute() / 'data'
+CACHE_PATH = DATA_PATH / 'univaf_raw'
+SUPPORTS_AWS_CLI = None
+
+
+def supports_aws_cli():
+    global SUPPORTS_AWS_CLI
+    if SUPPORTS_AWS_CLI is None:
+        result = subprocess.run(('which', 'aws'), capture_output=True)
+        SUPPORTS_AWS_CLI = result.returncode == 0
+
+    return SUPPORTS_AWS_CLI
+
+
+def s3_copy(source, destination):
+    subprocess.run(('aws', 's3', 'cp', source, destination))
+
+
+def download_file(bucket_path, destination_path, force=False):
+    if os.path.exists(destination_path) and not force:
+        return
+
+    Path(destination_path).parent.mkdir(parents=True, exist_ok=True)
+    if supports_aws_cli():
+        s3_uri = f's3://{UNIVAF_AWS_BUCKET}/{bucket_path}'
+        subprocess.run(('aws', 's3', 'cp', '--no-sign-request', s3_uri, destination_path))
+    else:
+        url = f'http://{UNIVAF_AWS_BUCKET}.s3.amazonaws.com/{bucket_path}'
+        print(f'Writing {url} to {destination_path}', file=sys.stderr)
+        with open(destination_path, 'wb') as f:
+            with urllib.request.urlopen(url) as remote:
+                f.write(remote.read())
+
+
+def log_file_name(log_type, date):
+    return f'{log_type}-{date}.ndjson.gz'
+
+
+def log_file_path(log_type, date):
+    return CACHE_PATH / log_file_name(log_type, date)
+
+
+def download_log_file(log_type, date):
```

Comment:
The signature on this function feels a little weird to me. As it is, I think you have to know exactly what it does in order to make much use of it.

Two change that might make things better that I'd propose:
 - Take the destination path so that it's obvious at the call site where the file's going to land
 - After downloading and saving, return the full on-disk destination path (and presumably use it rather than constructing the path elsewhere based on implicit knowledge of what `download_log_file` does)